### PR TITLE
TIP-135 WIP

### PR DESCRIPTION
**Pull request by Gorshtak**

### CODE REVIEW

File: src/components/Table/Table.tsx
Comment (Gorshtak) [inline src/components/Table/Table.tsx:1]: This is a new Table component that is used for leaderboard. It is written in a manner that rank will be displayed separated from the row and expandable part @vpoklopic

File: src/pages/Token/GamifiedStaking/RewardsV2/components/MobileRewardsView/MobileRewardsView.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+const MobileRewardsView: React.FC = () => {
+    const { t } = useTranslation();
+
+    const [learnMoreActive, setLearnMoreActive] = useState<boolean>(false);
+
+    return (
+        <Wrapper>
+            <Container>
+                <Header>{'Thales Staking Rewards'}</Header>
```
Comment (vpoklopic) [inline src/pages/Token/GamifiedStaking/RewardsV2/components/MobileRewardsView/MobileRewardsView.tsx:18]: Remove hardcoded text.

File: src/queries/token/usePointsBreakdownQuery.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    ),
+                    tradingVolume: formatCurrencyWithKey(
+                        USD_SIGN,
+                        bigNumberFormatter(tradingPoints) / bigNumberFormatter(tradingMultiplier)
+                    ),
+                    vaultsMultiplier: bigNumberFormatter(vaultsMultiplier),
+                    lpMultiplier: bigNumberFormatter(lpMultiplier),
+                    tradingMultiplier: bigNumberFormatter(tradingMultiplier),
+                    stakingMultiplier: formatCurrency(bigNumberFormatter(stakingMultiplier) + 1),
+                    vaultsPoints: formatCurrency(bigNumberFormatter(vaultsPoints)),
```
Comment (vpoklopic) [inline src/queries/token/usePointsBreakdownQuery.ts:101]: Formating of the currency should be done outside the query when we want to display data.

**Review by lotinasa — DISMISSED** [review]

File: src/queries/token/useStakingOverviewQuery.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            try {
+                const [period, bonusRewards, fixedPeriodReward] = await Promise.all([
+                    stakingThalesContract?.periodsOfStaking(),
+                    stakingThalesContract?.periodExtraReward(),
+                    stakingThalesContract?.fixedPeriodReward(),
+                ]);
+
+                return {
+                    period: Number(period),
+                    bonusRewards: formatCurrencyWithKey('', bigNumberFormatter(bonusRewards)),
```
Comment (vpoklopic) [inline src/queries/token/useStakingOverviewQuery.ts:31]: Same here for currency formating.

File: src/queries/token/useUserBaseRewards.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                    totalEscrowedRewards -
+                    totalEscrowBalanceNotIncludedInStaking;
+
+                const baseRewardsPool = bigNumberFormatter(contractStakingData.baseRewardsPool);
+
+                const baseRewards = (baseRewardsPool * (thalesStaked + escrowedBalance)) / totalStaked;
+
+                return {
+                    thalesStaked: formatCurrencyWithKey(THALES_CURRENCY, thalesStaked + escrowedBalance),
+                    totalStaked: formatCurrencyWithKey(THALES_CURRENCY, totalStaked),
```
Comment (vpoklopic) [inline src/queries/token/useUserBaseRewards.ts:67]: Same here for currency formating

File: src/types/governance.ts [hunk 5]
```diff
@@ -78,6 +78,27 @@ export type Staker = {
     ensName: string | null;
 };
 
+export type StakerContractLeaderboardData = {
```
Comment (vpoklopic) [inline src/types/governance.ts:81]: This is not part of governance, move to the right file.

**Review by vpoklopic — CHANGES_REQUESTED** [review]

File: src/queries/token/useUserBaseRewards.ts (hunk 4 above)
Comment (Gorshtak) [reply src/queries/token/useUserBaseRewards.ts:67]: Left for further iteration due to release.

File: src/queries/token/useStakingOverviewQuery.ts (hunk 3 above)
Comment (Gorshtak) [reply src/queries/token/useStakingOverviewQuery.ts]: Left for further iteration due to release.

**Review by vpoklopic — APPROVED** [review]

**Pull request merged**